### ISSUE TITLE
[FR] Add link to children in parent of multi-step

### ISSUE THREAD

**Issue by ndolev** [mlflow/mlflow#3862]
## Willingness to contribute

- [ ] Yes. I can contribute this feature independently.
- [ ] Yes. I would be willing to contribute this feature with guidance from the MLflow community.
- [X] No. I cannot contribute this feature at this time.

## Proposal Summary
In the MLflow UI, there is a link from child to parent steps but no link from parent to children. 
(In a few sentences, provide a clear, high-level description of the feature request)

## Motivation
- What is the use case for this feature?
To avoid having to return to the experiment page when exploring a particular run. 
- Why is this use case valuable to support for MLflow users in general?
This is valuable to anyone using the MLFlow UI to monitor and explore their runs.
- Why is this use case valuable to support for your project(s) or organization?
It makes it easier to explore the details of a multi-step run.
- Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)
It seems like a very simple addition but is lacking now. 

### What component(s), interfaces, languages, and integrations does this feature affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interfaces
- [ X] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Languages 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by garymm** [comment]
Duplicate of https://github.com/mlflow/mlflow/issues/835

**Comment by SebastianScherer88** [comment]
> Duplicate of [#835](https://github.com/mlflow/mlflow/issues/835)

True, but that duplicate issue has been closed without explanation 🤔
